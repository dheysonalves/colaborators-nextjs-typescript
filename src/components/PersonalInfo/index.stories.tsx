import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import PersonalInfo from '.'
import { FiFileText, FiPhoneCall, FiCalendar } from 'react-icons/fi'

export default {
	title: 'Composites/PersonalInfo',
	component: PersonalInfo,
	args: {
		title: 'Telefone',
		subtitle: '+55 82 2512 6627',
		IconName: FiPhoneCall,
	},
} as ComponentMeta<typeof PersonalInfo>

export const PhoneInfo: ComponentStory<typeof PersonalInfo> = (args) => (
	<PersonalInfo {...args} />
)

export const CPFInfo: ComponentStory<typeof PersonalInfo> = () => (
	<PersonalInfo title="CPF" subtitle="869 748 070 15" IconName={FiFileText} />
)

export const DateInfo: ComponentStory<typeof PersonalInfo> = () => (
	<PersonalInfo
		title="Nascimento"
		subtitle="29/11/1990"
		IconName={FiCalendar}
	/>
)
