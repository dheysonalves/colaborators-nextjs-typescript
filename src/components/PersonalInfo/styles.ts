import styled from '@emotion/styled'
import { BaseTypography, Paragraph } from '../Typography/base.index'

export const BorderContainer = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	border: 2px solid ${(props) => props.theme.light.neutral.neutral_1};
	border-radius: 8px;
	padding: 16px;
	background-color: ${(props) => props.theme.light.neutral.neutral_2};
`

export const TextContainer = styled.div`
	display: flex;
	flex-direction: column;
	margin-left: 8px;
`

export const Title = styled(BaseTypography)`
	color: ${(props) => props.theme.light.neutral.neutral_5};
	font-size: 12px;
	font-weight: 400;
`

export const Subtitle = styled(Paragraph)`
	color: ${(props) => props.theme.light.neutral.neutral_5};
	font-size: 14px;
	font-weight: 600;
`
