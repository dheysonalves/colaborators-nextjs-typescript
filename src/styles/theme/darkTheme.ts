export const theme = {
	dark: {
		brand: {
			blue: '#034AFD',
		},
		neutral: {
			background: '#F8FAF9',
			light: '#fff',
			black: '#34423D',
			neutral_2: '#F5FAF8',
			neutral_3: '#A3B8B0',
			neutral_5: '#587169',
		},
		primary: {
			default: '#22E0A1',
			green: '#B5F1DD',
		},
		feedback: {
			success: '#B5F1DD',
		},
		tertiary: {
			disabled: '#EAEFED',
		},
	},
}
