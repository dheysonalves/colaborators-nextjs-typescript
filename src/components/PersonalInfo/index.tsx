import React from 'react'
import { IconType } from 'react-icons'

import { BorderContainer, TextContainer, Title, Subtitle } from './styles'
import BulletIcon from '../BulletIcon/index'

interface PersonalInfoProps {
	title: string
	subtitle: string
	IconName: IconType
}

const PersonalInfo = ({ title, subtitle, IconName }: PersonalInfoProps) => {
	return (
		<BorderContainer>
			<BulletIcon Icon={IconName} iconSize={18} isSmall />
			<TextContainer>
				<Title tag="h2">{title}</Title>
				<Subtitle tag="p">{subtitle}</Subtitle>
			</TextContainer>
		</BorderContainer>
	)
}

export default PersonalInfo
