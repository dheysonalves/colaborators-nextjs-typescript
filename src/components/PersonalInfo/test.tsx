import React from 'react'

import { render, screen } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'
import { IconType } from 'react-icons'
import { FiFileText } from 'react-icons/fi'

import { theme } from '../../styles/theme/theme'

import PersonalInfo from '.'

const setup = (Icon: IconType, title: string, subtitle: string) => {
	const utils = render(
		<ThemeProvider theme={theme}>
			<PersonalInfo IconName={Icon} title={title} subtitle={subtitle} />,
		</ThemeProvider>,
	)
	return {
		...utils,
	}
}

describe('<PersonalInfo />', () => {
	it('should render the heading', () => {
		const { container } = setup(FiFileText, 'Telefone', '+55 82 2512 6627')

		expect(
			screen.getByRole('heading', { name: /Telefone/i }),
		).toBeInTheDocument()

		expect(container.firstChild).toMatchSnapshot()
	})
})
