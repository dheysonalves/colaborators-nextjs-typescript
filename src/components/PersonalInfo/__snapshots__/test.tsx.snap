// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonalInfo /> should render the heading 1`] = `
<div
  class="css-3y19al"
>
  <div
    class="css-1418cou"
  >
    <svg
      aria-label="icon-component"
      color="#465953"
      fill="none"
      height="18"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      style="color: rgb(70, 89, 83);"
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"
      />
      <polyline
        points="14 2 14 8 20 8"
      />
      <line
        x1="16"
        x2="8"
        y1="13"
        y2="13"
      />
      <line
        x1="16"
        x2="8"
        y1="17"
        y2="17"
      />
      <polyline
        points="10 9 9 9 8 9"
      />
    </svg>
  </div>
  <div
    class="css-110v27w"
  >
    <h2
      class="css-nfsw4d"
    >
      Telefone
    </h2>
    <p
      class="css-kqyjlq"
    >
      +55 82 2512 6627
    </p>
  </div>
</div>
`
